docs/explanations/architecture/styles.md

```diff
@@ -0,0 +1,143 @@
+## Styles in the block editor
+
+This document introduces the main concepts related to styles in the block editor. It points to the relevant reference guides and tutorials for readers to dig deeper into each one of the ideas presented. It's aimed to block authors and people working in the block editor project.
+
+1. HTML and CSS
+2. User-provided Block Styles
+    - From UI controls to HTML markup
+    - Block Supports API
+    - Current limits of the Block Supports API
+3. User-provided Global Styles
```

Comment:
Do we need a chapter about "Editor Styles" and how all these theme provided styles get loaded in the editor as well? (in addition to classic.css and reset.css) or is this specifically about frontend?

Reply:
I see this doc as a "map to the main subsystems" someone needs to be aware of, so not about the frontend. I've added a todo task in the issue description to consider this once the global styles section has a draft.

Reply:
Upon re-reading the whole document, I think the v1 doesn't need to cover this.

"How styles are enqueued into the editor" is a whole new topic that needs to cover a lot of ground by itself: `add_editor_style(...)`, `add_theme_support('editor-styles')`, `add_theme_support('wp-block-styles'), `reset.css`/`classic.css`, the iframe, `add_filter( 'enqueue_block_editor_assets', ...` and probably some other things I've forgotten. Some of this is already mentioned on [theme support](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/).